### PR TITLE
refactor: remove unused PoSt code

### PR DESCRIPTION
**Pull request by vmx**
There are several Proof-of-Spacetime which were never be used in production, remove those in order to lower the maintenance burden.

### CODE REVIEW

**Review by DrPeterVanNostrand — APPROVED** [review]
I'm fine with removing Election and Rational-PoSt, but I think that we should wait until after break to merge.

**Review by cryptonemo — APPROVED** [review]

**Pull request merged**
